<?xml version="1.0"?>
<robot name="oculus">
  <link name="oculus">
    <visual>
      <geometry>
        <mesh filename="package://pepper_surrogate/meshes/oculus_rift.dae" scale="0.0002 0.0002 0.0002"/>
      </geometry>
      <origin rpy="1.6 0 1.6" xyz="0 0 0"/>
    </visual>
  </link>
  <joint name="oculus_to_right_controller" type="revolute">
    <parent link="oculus"/>
    <child link="oculus_right_controller"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="oculus_right_controller">
    <visual>
      <geometry>
<!--         <cylinder length="0.01" radius="0.05"/> -->
        <mesh filename="package://pepper_surrogate/meshes/right_touch_controller.dae" scale="0.012 0.012 0.012"/>
      </geometry>
      <origin rpy="1.6 0 1.6" xyz="0 0 0"/>
    </visual>
  </link>
  <joint name="right_controller_to_right_vrhand" type="revolute">
    <parent link="oculus_right_controller"/>
    <child link="oculus_right_vrhand"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="oculus_right_vrhand">
    <visual>
      <geometry>
        <mesh filename="package://pepper_surrogate/meshes/Hand.dae" scale="0.05 0.05 0.05"/>
      </geometry>
      <origin rpy="0 0.8 3.14" xyz="0.03 0 0.02"/>
    </visual>
  </link>

  <joint name="oculus_to_left_controller" type="revolute">
    <parent link="oculus"/>
    <child link="oculus_left_controller"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="oculus_left_controller">
    <visual>
      <geometry>
<!--         <cylinder length="0.01" radius="0.05"/> -->
        <mesh filename="package://pepper_surrogate/meshes/left_touch_controller.dae" scale="0.012 0.012 0.012"/>
      </geometry>
      <origin rpy="1.6 0 1.6" xyz="0 0 0"/>
    </visual>
  </link>
  <joint name="left_controller_to_left_vrhand" type="revolute">
    <parent link="oculus_left_controller"/>
    <child link="oculus_left_vrhand"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1.0"/>
    <limit effort="2.666" lower="-1.56207" upper="-0.00872665" velocity="9.22756"/>
  </joint>
  <link name="oculus_left_vrhand">
    <visual>
      <geometry>
        <mesh filename="package://pepper_surrogate/meshes/Hand.dae" scale="0.05 -0.05 0.05"/>
      </geometry>
      <origin rpy="0 0.8 3.14" xyz="0.03 0 0.02"/>
    </visual>
  </link>
</robot>
